p2p/net/swarm/swarm_dial.go

```diff
@@ -44,6 +44,9 @@ var (
 // add loop back in Dial(.)
 const dialAttempts = 1
 
+// dial at most 12 peers concurrently
+const concurrentPeerDials = 12
```

Comment:
I chose 12 here because 12*10 (conccurent dials per peer) * 2 (reuseport) is around 1/4 of the available file descriptors on linux, and is still less than the 256 limit on OSX. Although maybe its a little too close to the OSX limit. Maybe we should set it at runtime with an OS check.